src/test/regress/sql/insert_select_into_local_table.sql

```diff
@@ -149,5 +174,238 @@ INSERT INTO non_dist_2 SELECT a, c FROM ref_table;
 SELECT * FROM non_dist_2 ORDER BY 1, 2;
 TRUNCATE non_dist_2;
 
+-- check issue https://github.com/citusdata/citus/issues/5858
+CREATE TABLE local_dest_table(
```

Comment:
It'd be great to have maybe some basic tests for where we add `local_dest_table` to the metadata? 

Then the final step becomes a COPY, which seems to work without issues, but seems like a good idea to given our investments to citus local tables